src/group_theory/nilpotent.lean

```diff
@@ -690,6 +691,46 @@ begin
   exact upper_central_series_eq_top_iff_nilpotency_class_le.mpr h,
 end
 
+section prod
+
+variables {G₁ G₂ : Type*} [group G₁] [group G₂]
+
+lemma lower_central_series_prod (n : ℕ):
+  lower_central_series (G₁ × G₂) n = (lower_central_series G₁ n).prod (lower_central_series G₂ n) :=
+begin
+  induction n with n ih,
+  { simp, },
+  { calc lower_central_series (G₁ × G₂) n.succ
+        = ⁅lower_central_series (G₁ × G₂) n, ⊤⁆  : rfl
+    ... = ⁅(lower_central_series G₁ n).prod (lower_central_series G₂ n), ⊤⁆ : by rw ih
+    ... = ⁅(lower_central_series G₁ n).prod (lower_central_series G₂ n), (⊤ : subgroup G₁).prod ⊤⁆ :
+      by simp
+    ... = ⁅lower_central_series G₁ n, (⊤ : subgroup G₁)⁆.prod ⁅lower_central_series G₂ n, ⊤⁆ :
+      general_commutator_prod_prod _ _ _ _
+    ... = (lower_central_series G₁ n.succ).prod (lower_central_series G₂ n.succ) : rfl }
+end
+
+/-- Products of nilpotent groups are nilpotent -/
+instance is_nilpotent_prod [is_nilpotent G₁] [is_nilpotent G₂] :
+  is_nilpotent (G₁ × G₂) :=
+begin
+  rw nilpotent_iff_lower_central_series,
+  refine ⟨max (group.nilpotency_class G₁) (group.nilpotency_class G₂), _ ⟩,
+  rw [lower_central_series_prod,
+    lower_central_series_eq_bot_iff_nilpotency_class_le.mpr (le_max_left _ _),
+    lower_central_series_eq_bot_iff_nilpotency_class_le.mpr (le_max_right _ _), bot_prod_bot],
+end
+
+/-- The nilpotency class of a product is the max of the nilpotency classes of the factors -/
+lemma nilpotency_class_prod [is_nilpotent G₁] [is_nilpotent G₂] :
+  group.nilpotency_class (G₁ × G₂) = max (group.nilpotency_class G₁) (group.nilpotency_class G₂) :=
```

Comment:
I've just closed my gitpod session so I can't test but isn't the `group` namespace open here?

Can we write `nilpotency_class` instead of `group.nilpotency_class`?

Reply:
I'll check

Reply:
Nope, `group` is not open in this file, it seems

Reply:
Thanks for checking. I wouldn't spend any more time on this.

FWIW, looking at the file, I think the `group` namespace is open but the statement `def group.nilpotency_class` occurs below `open group` and there is this horrible feature whereby `open group` thus doesn't pick it up.

Reply:
“feature”? :-)